pkg/gomod/modules_test.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gomod
+
+import (
+	"testing"
+)
+
+func TestModule(t *testing.T) {
+	tests := map[string]struct {
```

Comment:
please don't do `map` just fold the name into the struct
```
tests := []struct {
 name string..
...
}
```

You'll also get better formatting of the cases.

Reply:
strong disagree

Reply:
Tell me how i this better?

Reply:
this requires that each name of each test is unique, giving you a signal you did not just copy/paste a row in the table.

Reply:
Literally never happened to me :-)